sparse/coo.py

```diff
@@ -2149,101 +2133,173 @@ def _elemwise_binary(func, self, other, *args, **kwargs):
     return COO(coords, data, shape=result_shape, has_duplicates=False)
 
 
-def _elemwise_binary_self_dense(func, self, other, *args, **kwargs):
-    assert isinstance(self, np.ndarray)
-    assert isinstance(other, COO)
+def _match_coo(*args, **kwargs):
+    """
+    Matches the coordinates for any number of input :obj:`COO` arrays.
+    Equivalent to "sparse" broadcasting for all arrays.
+
+    Parameters
+    ----------
+    args : Tuple[COO]
+        The input :obj:`COO` arrays.
+    return_midx : bool
+        Whether to return matched indices or matched arrays. Matching
+        only supported for two arrays. ``False`` by default.
+    cache : dict
+        Cache of things already matched. No cache by default.
+
+    Returns
+    -------
+    matched_idx : List[ndarray]
+        The indices of matched elements in the original arrays. Only returned if
+        ``return_midx`` is ``True``.
+    matched_arrays : List[COO]
+        The expanded, matched :obj:`COO` objects. Only returned if
+        ``return_midx`` is ``False``.
+    """
+    return_midx = kwargs.pop('return_midx', False)
+    cache = kwargs.pop('cache', None)
+
+    if kwargs:
+        raise ValueError('Unknown kwargs %s' % kwargs.keys())
+
+    if return_midx and (len(args) != 2 or cache is not None):
+        raise NotImplementedError('Matching only supported for two args, and no cache.')
```

Comment:
Do we still need this option?

Reply:
No, we don't. I'm not omniscient, so I went ahead and added this check in case someone tried to trigger caching on `return_midx` (which we don't cache, it's never repeated); or tried to match indices for `len(args) != 2` (I'm not sure if we'll need this in the future, but we might, and it's useful to err rather than have it return incorrect results).